classes/class-p4-post-campaign.php

```diff
@@ -315,98 +315,163 @@ private static function campaign_field(
 
-			$theme = empty( $meta['theme'] )
-								? 'default'
-								: $meta['theme'];
+			// TODO: Use wp_safe_remote_get?
+			// TODO: Handle errors
+			$theme_json = json_decode(
+				file_get_contents( __DIR__ . '/../campaign_themes/' . $theme . '.json' ),
+				true
+			);
 
-			$header_primary_font = empty( $meta['campaign_header_primary'] )
-															? $campaigns_font_map[ $theme ]
-															: $meta['campaign_header_primary'];
+			$defaults = self::get_theme_defaults( $theme_json );
 
-			$body_font = empty( $meta['campaign_body_font'] )
-										? $campaigns_font_map[ $theme ]
-										: $meta['campaign_body_font'];
+			// Replace the defaults with the campaign options where applicable
+			// 'array_intersect_key' removes the keys from $meta that are not present in the $defaults.
+			// 'array_filter' removes empty (falsey?) values from the resulting array.
+			$css_vars = array_filter( array_intersect_key( $meta, $defaults ) );
 
-			// TODO: Remove this special case.
-			$header_primary_font = str_replace( 'Montserrat_Light', 'Montserrat', $header_primary_font );
-			$body_font           = str_replace( 'Montserrat_Light', 'Montserrat', $body_font );
+			$css_vars = array_merge( $css_vars, self::get_footer_theme( $meta ) );
+			$css_vars = array_merge( $css_vars, self::get_passive_button_color( $meta, $defaults ) );
+			$css_vars = self::replace_font_aliases( $css_vars );
 
-			$footer_theme = $meta['campaign_footer_theme'] ?? null;
+			return $css_vars;
+		}
 
-			if ( 'white' === $footer_theme ) {
-				$default_footer_links_color = $meta['campaign_nav_color'] ? $meta['campaign_nav_color'] : '#1A1A1A';
-				$footer_links_color         = $meta['footer_links_color'] ? $meta['footer_links_color'] : $default_footer_links_color;
-				$footer_color               = '#FFFFFF';
-			} else {
-				switch ( ( $meta['campaign_logo_color'] ?? null ) ) {
-					case 'dark':
-						$footer_links_color = '#1A1A1A';
-						break;
-					case 'green':
-						$footer_links_color = '#2caf4e';
-						break;
-					default:
-						$footer_links_color = '#FFFFFF';
-				}
+		/**
+		 * DEPRECATE: This function replaces a font names in the CSS variables.
+		 *
+		 * @param array $css_vars The array containing the CSS variables.
+		 * @return array The variables for the footer theme.
+		 */
+		public static function replace_font_aliases( array $css_vars ): array {
+			// TODO: Remove these special cases.
+			if ( isset( $css_vars['campaign_header_primary'] ) ) {
+				$css_vars['campaign_header_primary'] = str_replace( 'Montserrat_Light', 'Montserrat', $css_vars['campaign_header_primary'] );
+			}
 
-				$footer_color = $meta['campaign_nav_color'] ?? null;
+			if ( isset( $css_vars['campaign_body_font'] ) ) {
+				$css_vars['campaign_body_font'] = str_replace( 'Montserrat_Light', 'Montserrat', $css_vars['campaign_body_font'] );
 			}
 
+			return $css_vars;
+		}
+
+		/**
+		 * DEPRECATE: Get the passive button color based on the meta settings.
```

Comment:
You can use the `@deprecated` annotation, then IDE's will highlight this wherever it's used.

Reply:
Also best to include explanation why it's deprecated and what the alternative is.